### ISSUE TITLE
bug on LSTM

### ISSUE THREAD

**Issue by njuhang** [quic/aimet#4038]
I followed instructions in lpai4.x sdk, CRNN example. The model is defined as:

    class CRNN(nn.Module): 
        def __init__(self, num_classes=12, dropout_prob=0.1, is_bidirectional=False):
            super().__init__()
            self.n_maps = 48
            self.fc_maps = 84

            # input NCHW (1,1,10,49)
            self.conv0 = nn.Conv2d(1, self.n_maps, (4, 10), padding=0, bias=True, stride=2)
            self.relu0 = nn.ReLU()
            self.lstm = nn.LSTM(input_size=192, hidden_size=64, num_layers=2, bidirectional=is_bidirectional)
            if is_bidirectional:
                self.fc1 = nn.Linear(2560, self.fc_maps, bias=True)
            else:
                self.fc1 = nn.Linear(1280, self.fc_maps, bias=True)
            self.relu1 = nn.ReLU()
            self.fc2 = nn.Linear(self.fc_maps, num_classes, bias=True)

      def forward(self, x):
          # See note [TorchScript super()]
          x = self.relu0(self.conv0(x))
          x = x.permute(3, 0, 1, 2)
          x = torch.reshape(x, (x.shape[0], -1, x.shape[2]*x.shape[3]))
          x, hidden = self.lstm(x)
          x = torch.transpose(x, 0, 1)
          x = torch.flatten(x, 1)
          x = self.relu1(self.fc1(x))
          x = self.fc2(x)
  
          return x
  
    def crnn_small(num_classes=12):
        model = CRNN(num_classes=num_classes, is_bidirectional=False)
        return model
    
    def crnn_bidir_small(num_classes=12):
        model = CRNN(num_classes=num_classes, is_bidirectional=True)
        return model

And I use aimet to do quantization, but there is a bug on it:
2025-10-30 16:07:13,165 - Utils - INFO - successfully created onnx model with 29/48 node names updated
2025-10-30 16:07:13,178 - Quant - WARNING - number of output quantizers: 3 available for layer: lstm doesn't match with number of output tensors: 1
2025-10-30 16:07:13,179 - Quant - ERROR - Param tensor {lstm.weight_ih_l0} not found in valid param set
2025-10-30 16:07:13,180 - Quant - ERROR - Param tensor {lstm.weight_hh_l0} not found in valid param set
2025-10-30 16:07:13,180 - Quant - ERROR - Param tensor {lstm.bias_ih_l0} not found in valid param set
2025-10-30 16:07:13,180 - Quant - ERROR - Param tensor {lstm.bias_hh_l0} not found in valid param set
2025-10-30 16:07:13,180 - Quant - ERROR - Param tensor {lstm.weight_ih_l1} not found in valid param set
2025-10-30 16:07:13,180 - Quant - ERROR - Param tensor {lstm.weight_hh_l1} not found in valid param set
2025-10-30 16:07:13,181 - Quant - ERROR - Param tensor {lstm.bias_ih_l1} not found in valid param set
2025-10-30 16:07:13,181 - Quant - ERROR - Param tensor {lstm.bias_hh_l1} not found in valid param set
2025-10-30 16:07:13,183 - Quant - INFO - Layers excluded from quantization: []
2025-10-30 16:07:13,185 - Quant - WARNING - [31;21mQuantsim export will stop exporting encodings for saving and loading in a future AIMET release.

It seems that parameters name in LSTM layer is not capable with corresponding .onnx name.

Other key code are listed below:

    sim = QuantizationSimModel(crnn,
                                   quant_scheme=quant_scheme,
                                   dummy_input= dummy_input,
                                   default_param_bw=w_bits,
                                   default_output_bw=act_bits)
    sim.compute_encodings(model_eval, forward_pass_callback_args=(test_loader, args.num_batches))
    sim.export(dir_name, file_name, dummy_1, onnx_export_args=OnnxExportApiArgs(opset_version=14))

**Comment by njuhang** [comment]
btw, my environment is 
aimet                         0.0.1
aimet-torch               2.17.0
torch                         2.8.0
onnx                         1.19.1

**Comment by quic-kyunggeu** [comment]
Hi @njuhang, thank you for the detailed repro script.
This is a known issue that `sim.export` can't handle LSTM/RNN/GRU.
Instead, I recommend using `sim.onnx.export` or [`aimet_torch.onnx.export`](https://quic.github.io/aimet-pages/releases/latest/apiref/torch/onnx.html#apiref-torch-onnx) instead of `sim.export`, which is a soon-to-be-deprecated API anyway.

```py
import aimet_torch

def main():
    crnn = crnn_small()
    x = torch.randn(1,1,10,49)
    sim = aimet_torch.QuantizationSimModel(crnn, x)
    sim.compute_encodings(lambda model: model(x))

    # Takes the same set of arguments as torch.onnx.export
    aimet_torch.onnx.export(sim.model, x, "crnn_small_quantized.onnx",
                            input_names=["input"], output_names=["output"],
                            opset_version=21)

    # Takes the same set of arguments as torch.onnx.export,
    # except it doesn't take model as the first input (`sim` is already the model itself)
    sim.onnx.export(x, "crnn_small.onnx", input_names=["input"], output_names=["output"])


if __name__ == "__main__":
    main()
```

**Comment by quic-kyunggeu** [comment]
Also some disclaimers you should note...
Today, AIMET does **NOT** provide an elegant support for LSTM / LPAI.
AIMET does quantize LSTM, but most likely it's not going to be deployable to LPAI out-of-the-box.
We are currently in the active process of improving it to support LSTM / LPAI more seamlessly.

**Comment by njuhang** [comment]
Roger that, I'll make a validation on it. 
thank you

**Comment by quic-kyunggeu** [comment]
@njuhang See also https://github.com/quic/aimet/issues/4044#issuecomment-3508493951.
(summary: Our goal is to restore LSTM support for _**eNPU**_ by early next year)
